sdk/core/Azure.Core/src/ResponseError.cs

```diff
@@ -189,5 +197,48 @@ internal void Append(StringBuilder builder, bool includeRaw)
                 builder.Append(_element.GetRawText());
             }
         }
+
+        /// <inheritdoc />
+        public void Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+        {
+            throw new NotSupportedException(NotSupportedExceptionMessage);
```

Comment:
JsonConverter throws in Serialize call (L147) so kept the same behavior in Write